src/models/Call.ts

```diff
@@ -566,18 +574,101 @@ export class ElementCall extends Call {
+            if (hasCallStarted) {
+                params.append("intent", ElementCallIntent.JoinExistingDM);
+                params.append("preload", "false");
+            } else {
+                params.append("intent", ElementCallIntent.StartCallDM);
+                params.append("preload", "false");
+            }
+        } else {
+            params.append("sendNotificationType", "notification");
+            if (hasCallStarted) {
+                params.append("intent", ElementCallIntent.JoinExisting);
+                params.append("preload", "false");
+            } else {
+                params.append("intent", ElementCallIntent.StartCall);
+                params.append("preload", "false");
+            }
+        }
+    }
 
-        const elementCallUrl = SettingsStore.getValue("Developer.elementCallUrl");
-        if (elementCallUrl) url = new URL(elementCallUrl);
+    /**
+     * Calculate the correct analytics parameters for an Element Call room. Paarameters
+     * will be applied to the `params` instance.
+     *
+     * @param params Existing URL parameters
+     * @param client The current client.
+     */
+    private static appendAnalyticsParams(params: URLSearchParams, client: MatrixClient): void {
+        const posthogConfig = SdkConfig.get("posthog");
+        if (!posthogConfig || PosthogAnalytics.instance.getAnonymity() === Anonymity.Disabled) {
+            return;
+        }
+
+        const accountAnalyticsData = client.getAccountData(PosthogAnalytics.ANALYTICS_EVENT_TYPE)?.getContent();
+        // The analyticsID is passed directly to element call (EC) since this codepath is only for EC and no other widget.
+        // We really don't want the same analyticID's for the EC and EW posthog instances (Data on posthog should be limited/anonymized as much as possible).
+        // This is prohibited in EC where a hashed version of the analyticsID is used for the actual posthog identification.
+        // We can pass the raw EW analyticsID here since we need to trust EC with not sending sensitive data to posthog (EC has access to more sensible data than the analyticsID e.g. the username)
+        const analyticsID: string = accountAnalyticsData?.pseudonymousAnalyticsOptIn ? accountAnalyticsData?.id : "";
+
+        params.append("analyticsID", analyticsID); // Legacy, deprecated in favour of posthogUserId
+        params.append("posthogUserId", analyticsID);
+        params.append("posthogApiHost", posthogConfig.api_host);
+        params.append("posthogApiKey", posthogConfig.project_api_key);
+
+        // We gate passing sentry behind analytics consent as EC shares data automatically without user-consent,
+        // unlike EW where data is shared upon an intentional user action (rageshake).
+        const sentryConfig = SdkConfig.get("sentry");
+        if (sentryConfig) {
+            params.append("sentryDsn", sentryConfig.dsn);
+            params.append("sentryEnvironment", sentryConfig.environment ?? "");
```

Comment:
should this also be refactored the same we refactored skipLobby?

Reply:
Oh the lines seem to be not correct.

Reply:
I think your last commit just jumped in before i pressed send on the review.

I am talking about the `returnToLobby: "$returnToLobby",` prop

Reply:
sorted, I think.